pkg/util/v1beta1/katibconfig/config.go

```diff
@@ -36,6 +36,11 @@ import (
 type SuggestionConfig struct {
 	Image                     string                           `json:"image"`
 	ImagePullPolicy           corev1.PullPolicy                `json:"imagePullPolicy,omitempty"`
+	Command                   []string                         `json:"command,omitempty"`
+	Args                      []string                         `json:"args,omitempty"`
+	WorkingDir                string                           `json:"workingDir,omitempty"`
+	EnvFrom                   []corev1.EnvFromSource           `json:"envFrom,omitempty"`
+	Env                       []corev1.EnvVar                  `json:"env,omitempty"`
```

Comment:
Maybe we could start adding more fields to SuggestionConfig with `corev1.Container{}` field, similar to [`PodSpec` API in K8s](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L3105) ? I think you can use inline settings to avoid modification in existing Katib config YAML:
```go
 v1.Container `json:",inline"`
 ```

I guess, at the end we want to give user ability to modify the whole Suggestion's Deployment spec via Katib config, but it requires much more changes.

WDYT @tenzen-y @johnugeorge @fischor ?

Reply:
> Maybe we could start adding more fields to SuggestionConfig with corev1.Container{} field, similar to [PodSpec API in K8s](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L3105) ? I think you can use inline param to avoid modification in existing Katib config YAML

sgtm

> I guess, at the end we want to give user ability to modify the whole Suggestion's Deployment spec via Katib config, but it requires much more changes.

I think so too. Maybe, to support customizing Tolerations, imagePullSecret and so on, we want to support customizing PodSpec for suggestion service deployment.

Reply:
Yes, inlining `corev1.Container` should also be doable.  
Certain fields of the container will be set by the katib-Controller, like some ports, the liveness and readiness probes and possibly the "suggestion-volume". How to handle these?

> I think so too. Maybe, to support customizing Tolerations, imagePullSecret and so on, we want to support customizing PodSpec for suggestion service deployment.

To support his, how about specifying a default deployment in the suggestion config (instead of just the container).

Reply:
For namespace specific specific suggestion deployments, we can allow to have a configmap in a user namespace(where experiment is run). If suggestion spec is provided in the configmap of the user namespace, it can override the global katib config.  
Ref: https://github.com/kubeflow/katib/issues/1976

Reply:
> Certain fields of the container will be set by the katib-Controller, like some ports, the liveness and readiness probes and possibly the "suggestion-volume". How to handle these? Overwrite or not?

@fischor I think if parameters are set by the user we should use them, otherwise use the default values.

> To support his, how about specifying a default deployment in the suggestion config (instead of just the container)?.

We might want to set `spec.template.spec` for our Suggestion config, as I mentioned here: https://github.com/kubeflow/katib/issues/1737#issuecomment-974245081, but that requires much more changes since we need to modify manifests.

>  If suggestion spec is provided in the configmap of the user namespace, it can override the global katib config.

@johnugeorge That might be the next step. We need to have discussion what is the best way to give user ability to set experiment-scoped Katib Config.